python-interpreter-builder/scripts/build-python-3.5.sh

```diff
@@ -1,21 +1,132 @@
 #!/bin/bash
 
-set -e
+set -euo pipefail
+set -x
 
 # Get the source
 mkdir -p /opt/sources
 cd /opt/sources
-wget -nv https://www.python.org/ftp/python/3.5.2/Python-3.5.2.tgz
-tar xzf Python-3.5.2.tgz
+wget --no-verbose https://www.python.org/ftp/python/3.5.3/Python-3.5.3.tgz
+# SHA-256 generated via `shasum -a 256 [file]`
+shasum --check <<EOF
+d8890b84d773cd7059e597dbefa510340de8336ec9b9e9032bf030f19291565a  Python-3.5.3.tgz
+EOF
+tar xzf Python-3.5.3.tgz
 
 # Build
-cd Python-3.5.2
-# Use /opt/python{X}.{Y} for the prefix.
-./configure --prefix=/opt/python3.5 --with-lto
-# Explicitly build the profile-guided-optimized interpreter
+cd Python-3.5.3
+
+# Explanation of flags:
+#
+# Noteworthy Debian options we _don't_ use:
+#
+# --enable-shared
+#   This is complicated to get right, and we don't expect our
+#   customers to embed Python in a native code application.  There is
+#   also a noteworthy interaction with 'make altinstall'
+#   (https://bugs.python.org/issue27685)
+# --without-ensurepip
+#   Debian unbundles pip for their own reasons
+# CFLAGS=-fdebug-prefix-map
+#   Unnecessary in our build environment
+#
+#
+# Flags that we _do_ use:
+# (Debian) means it was taken from Debian build rules.
+#
+# --build
+#   (Debian)
+# --enable-ipv6
```

Comment:
FYI, this makes sure ipv6 support is compiled in. If you don't pass the argument, an autoconf compile check is used, which in the past sometimes gave the wrong answer. (It ended up checking kernel features rather than API support.) I don't know if that's still the case, but better safe than sorry.